components/ButtonBar/examples/ButtonBarCustomizedFontExample.m

```diff
@@ -0,0 +1,117 @@
+
+  MDCButtonBar *buttonBar = [[MDCButtonBar alloc] init];
+
+  // MDCButtonBar ignores the style of UIBarButtonItem.
+  UIBarButtonItemStyle ignored = UIBarButtonItemStyleDone;
+
+  UIBarButtonItem *actionItem =
+      [[UIBarButtonItem alloc] initWithTitle:@"Disable"
+                                       style:ignored
+                                      target:self
+                                      action:@selector(didTapActionButton:)];
+  UIBarButtonItem *secondActionItem =
+      [[UIBarButtonItem alloc] initWithTitle:@"Customization"
+                                       style:ignored
+                                      target:self
+                                      action:@selector(didTapActionButton:)];
+
+  NSArray *items = @[ actionItem, secondActionItem ];
+
+  // Set the title text attributes before assigning to buttonBar.items
+  // because of https://github.com/material-components/material-components-ios/issues/277
+  for (UIBarButtonItem *item in items) {
+    [item setTitleTextAttributes:@{ NSForegroundColorAttributeName : [UIColor greenColor] }
+                        forState:UIControlStateNormal];
+  }
+
+  buttonBar.items = items;
+
+  // MDCButtonBar's sizeThatFits gives a "best-fit" size of the provided items.
+  CGSize size = [buttonBar sizeThatFits:self.view.bounds.size];
+  CGFloat x = (self.view.bounds.size.width - size.width) / 2;
+  CGFloat y = self.view.bounds.size.height / 2 - size.height;
+  buttonBar.frame = CGRectMake(x, y, size.width, size.height);
+  buttonBar.autoresizingMask =
+      (UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleBottomMargin |
+       UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin);
+  [self.view addSubview:buttonBar];
+
+  // Ensure that the controller's view isn't transparent.
+  self.view.backgroundColor = [UIColor colorWithWhite:0.9f alpha:1.0f];
+}
+
+#pragma mark - User actions
+
+- (void)didTapActionButton:(id)sender {
+  NSLog(@"Did tap action item: %@", sender);
+
+  // Disable CUstomization
+  [[MDCButtonBarButton appearance] setTitleFont:nil forState:UIControlStateNormal];
+}
+
+@end
+
+@implementation ButtonBarCustomizedFontExample (CatalogByConvention)
+
++ (NSArray *)catalogBreadcrumbs {
+  return @[ @"Button Bar", @"Button Bar (Customized)" ];
+}
+
++ (BOOL)catalogIsPrimaryDemo {
+  return NO;
+}
+
++ (NSString *)catalogDescription {
+  return @"A Button Bar with a customized font.";
+}
+
++ (BOOL)catalogIsPresentable {
+  return YES;
+}
+
+@end
+
+#pragma mark - Typical application code (not Material-specific)
+
+@implementation ButtonBarCustomizedFontExample (GeneralApplicationLogic)
+
+- (id)init {
+  self = [super init];
+  if (self) {
+    self.title = @"Button Bar";
+
+    UIFont *customFont = [UIFont fontWithName:@"American Typewriter" size:10.0f];
+    NSAssert(customFont, @"Unable to instantiate font");
+    [[MDCButtonBarButton appearance] setTitleFont:customFont forState:UIControlStateNormal];
```

Comment:
Just so I understand correctly, opening this example will change the font for all ButtonBar buttons throughout the app?

Reply:
Correct.  However, if you actually click on one of the button bar buttons, it will set the appearance proxy back to nil / default.

Reply:
Got it. Added a comment above in the action target to recommend changing the instances as well as UIAppearance.